tools/downloader/src/open_model_zoo/model_tools/converter.py

```diff
@@ -104,103 +104,131 @@ def main():
+                string.Template(arg).substitute(template_variables)
+                for arg in model.mo_args]
 
-            if not args.dry_run:
-                reporter.print(flush=True)
+            for model_precision in sorted(model_precisions):
+                data_type = model_precision.split('-')[0]
+                mo_cmd = [str(args.python), '--', str(mo_path),
+                    '--framework={}'.format(model_format),
+                    '--data_type={}'.format(data_type),
+                    '--output_dir={}'.format(output_dir / model.subdirectory / model_precision),
+                    '--model_name={}'.format(model.name),
+                    *expanded_mo_args, *extra_mo_args]
 
-                if not reporter.job_context.subprocess(mo_cmd):
-                    return False
+                reporter.print_section_heading('{}Converting {} to IR ({})',
+                    '(DRY RUN) ' if args.dry_run else '', model.name, model_precision)
 
-            reporter.print()
+                reporter.print('Conversion command: {}', _common.command_string(mo_cmd))
 
-        return True
+                if not args.dry_run:
+                    reporter.print(flush=True)
+
+                    telemetry.send_event('md', 'converter_precision', model_precision)
 
-    reporter = _common.Reporter(_common.DirectOutputContext())
+                    if not reporter.job_context.subprocess(mo_cmd):
+                        telemetry.send_event('md', 'converter_failed_models', model.name)
+                        telemetry.send_event('md', 'converter_error', str({'error': 'mo-failed', 'model': model.name}))
+                        return False
 
-    if args.jobs == 1 or args.dry_run:
-        results = [convert(reporter, model) for model in models]
-    else:
-        results = _common.run_in_parallel(args.jobs,
-            lambda context, model: convert(_common.Reporter(context), model),
-            models)
+                reporter.print()
 
-    failed_models = [model.name for model, successful in zip(models, results) if not successful]
+            return True
 
-    if failed_models:
-        reporter.print('FAILED:')
-        for failed_model_name in failed_models:
-            reporter.print(failed_model_name)
-        sys.exit(1)
+        reporter = _common.Reporter(_common.DirectOutputContext())
+
+        if args.jobs == 1 or args.dry_run:
+            results = [convert(reporter, model) for model in models]
+        else:
+            results = _common.run_in_parallel(args.jobs,
+                lambda context, model: convert(_common.Reporter(context), model),
+                models)
+
+        failed_models = [model.name for model, successful in zip(models, results) if not successful]
+
+        if failed_models:
+            reporter.print('FAILED:')
+            for failed_model_name in failed_models:
+                reporter.print(failed_model_name)
+            telemetry.send_event('md', 'converter_result', 'fail')
```

Comment:
The exit status reporting is common between tools, so I would suggest to unify it. If you implement my suggestion about `telemetry_session` above, then it'll be pretty easy to do by adding some exception handling to the `try` block:

```python
except SystemExit as e:
    telemetry.send_event('md', f'{tool}_result', 'failure' if e.code else 'success')
    raise
except BaseException:
    telemetry.send_event('md', f'{tool}_result', 'exception')
    raise
else:
    telemetry.send_event('md', f'{tool}_result', 'success')
```

(I also added the reporting of exceptions, since I think it could be useful to log them, while distinguishing them from normal exits.)

`tool` would have to be passed in as a parameter. That said, I don't know how the `app_name` parameter is used by the telemetry library - maybe it would be sufficient to distinguish between tools?

Reply:
Done